CobiGen SOAP problem

I’m making a Devonfw PoC End-To-End in order to test the CobiGen features in order to generate REST, SOAP and Angular templates.

I'm using Devonfw 2.3.1

The PoC is very simple.
Starting from an Entity called EmployeeEntity:

```java
package com.capgemini.poc.devonita.general.dataaccess.api;

import javax.persistence.Entity;

@Entity
@javax.persistence.Table(name = "EMPLOYEE" )
public class EmployeeEntity {

  @Column(name = "EMPLOYEEID")
  @GeneratedValue(strategy = GenerationType.IDENTITY)
  private Long employeeId; 

  @Column(name = "NAME")
  private String name;

  @Column(name = "ROLE")
  private String role;

  @Column(name = "EMAIL")
  private String email;

}
```

I want to generate with CobiGen:
•	REST services
•	SOAP CRUD services 
•	Angular UI CRUD

If I select JUST Rest services, the Backend is successfully generated, compiled, deployed and tested at runtime.


BUT if I also check SOAP (I need to generate BOTH Rest and Soap services) 
 

I have the following error when I start the server:

```
[D: 2018-04-06 16:46:46,780] [P: INFO ] [C: ] [T: localhost-startStop-1] [L: com.capgemini.poc.devonita.general.service.impl.config.ServletInitializer] - [M: Starting ServletInitializer on LIT001774 with PID 8784 (D:\Devon-dist_2.3.1\workspaces\main\.metadata\.plugins\org.eclipse.wst.server.core\tmp0\wtpwebapps\devonita-server\WEB-INF\lib\devonita-core-0.0.1-SNAPSHOT.jar started by SROSSINI in D:\Devon-dist_2.3.1\software\eclipse)]  
[D: 2018-04-06 16:46:46,800] [P: DEBUG] [C: ] [T: localhost-startStop-1] [L: com.capgemini.poc.devonita.general.service.impl.config.ServletInitializer] - [M: Running with Spring Boot v1.5.3.RELEASE, Spring v4.3.8.RELEASE]  
[D: 2018-04-06 16:46:46,801] [P: INFO ] [C: ] [T: localhost-startStop-1] [L: com.capgemini.poc.devonita.general.service.impl.config.ServletInitializer] - [M: The following profiles are active: h2mem]  
[D: 2018-04-06 16:46:47,048] [P: INFO ] [C: ] [T: localhost-startStop-1] [L: org.springframework.boot.context.embedded.AnnotationConfigEmbeddedWebApplicationContext] - [M: Refreshing org.springframework.boot.context.embedded.AnnotationConfigEmbeddedWebApplicationContext@460a8c06: startup date [Fri Apr 06 16:46:47 CEST 2018]; root of context hierarchy]  
[D: 2018-04-06 16:46:47,341] [P: DEBUG] [C: ] [T: background-preinit] [L: org.jboss.logging] - [M: Logging Provider: org.jboss.logging.Slf4jLoggerProvider found via system property]  
[D: 2018-04-06 16:46:47,348] [P: INFO ] [C: ] [T: background-preinit] [L: org.hibernate.validator.internal.util.Version] - [M: HV000001: Hibernate Validator 5.3.5.Final]  
[D: 2018-04-06 16:46:49,703] [P: INFO ] [C: ] [T: localhost-startStop-1] [L: org.springframework.beans.factory.xml.XmlBeanDefinitionReader] - [M: Loading XML bean definitions from class path resource [META-INF/cxf/cxf.xml]]  
[D: 2018-04-06 16:46:49,841] [P: INFO ] [C: ] [T: localhost-startStop-1] [L: org.springframework.beans.factory.support.DefaultListableBeanFactory] - [M: Overriding bean definition for bean 'cxf' with a different definition: replacing [Root bean: class [null]; scope=; abstract=false; lazyInit=false; autowireMode=3; dependencyCheck=0; autowireCandidate=true; primary=false; factoryBeanName=serviceConfig; factoryMethodName=springBus; initMethodName=null; destroyMethodName=(inferred); defined in class path resource [com/capgemini/poc/devonita/general/service/impl/config/ServiceConfig.class]] with [Generic bean: class [org.apache.cxf.bus.spring.SpringBus]; scope=; abstract=false; lazyInit=false; autowireMode=0; dependencyCheck=0; autowireCandidate=true; primary=false; factoryBeanName=null; factoryMethodName=null; initMethodName=null; destroyMethodName=shutdown; defined in class path resource [META-INF/cxf/cxf.xml]]]  
[D: 2018-04-06 16:46:50,319] [P: INFO ] [C: ] [T: localhost-startStop-1] [L: org.springframework.beans.factory.support.DefaultListableBeanFactory] - [M: Overriding bean definition for bean 'servletRegistrationBean' with a different definition: replacing [Root bean: class [null]; scope=; abstract=false; lazyInit=false; autowireMode=3; dependencyCheck=0; autowireCandidate=true; primary=false; factoryBeanName=serviceConfig; factoryMethodName=servletRegistrationBean; initMethodName=null; destroyMethodName=(inferred); defined in class path resource [com/capgemini/poc/devonita/general/service/impl/config/ServiceConfig.class]] with [Root bean: class [null]; scope=; abstract=false; lazyInit=false; autowireMode=3; dependencyCheck=0; autowireCandidate=true; primary=false; factoryBeanName=io.oasp.module.cxf.common.impl.server.CxfServerAutoConfiguration; factoryMethodName=servletRegistrationBean; initMethodName=null; destroyMethodName=(inferred); defined in class path resource [io/oasp/module/cxf/common/impl/server/CxfServerAutoConfiguration.class]]]  
[D: 2018-04-06 16:46:50,320] [P: INFO ] [C: ] [T: localhost-startStop-1] [L: org.springframework.beans.factory.xml.XmlBeanDefinitionReader] - [M: Loading XML bean definitions from class path resource [META-INF/cxf/cxf.xml]]  
[D: 2018-04-06 16:46:50,365] [P: INFO ] [C: ] [T: localhost-startStop-1] [L: org.springframework.beans.factory.support.DefaultListableBeanFactory] - [M: Overriding bean definition for bean 'jaxRsServer' with a different definition: replacing [Root bean: class [null]; scope=; abstract=false; lazyInit=false; autowireMode=3; dependencyCheck=0; autowireCandidate=true; primary=false; factoryBeanName=serviceConfig; factoryMethodName=jaxRsServer; initMethodName=null; destroyMethodName=(inferred); defined in class path resource [com/capgemini/poc/devonita/general/service/impl/config/ServiceConfig.class]] with [Root bean: class [null]; scope=; abstract=false; lazyInit=false; autowireMode=3; dependencyCheck=0; autowireCandidate=true; primary=false; factoryBeanName=io.oasp.module.cxf.common.impl.server.rest.CxfServerRestAutoConfiguration; factoryMethodName=jaxRsServer; initMethodName=null; destroyMethodName=(inferred); defined in class path resource [io/oasp/module/cxf/common/impl/server/rest/CxfServerRestAutoConfiguration.class]]]  
[D: 2018-04-06 16:46:50,367] [P: INFO ] [C: ] [T: localhost-startStop-1] [L: org.springframework.beans.factory.support.DefaultListableBeanFactory] - [M: Overriding bean definition for bean 'jacksonJsonProvider' with a different definition: replacing [Root bean: class [null]; scope=; abstract=false; lazyInit=false; autowireMode=3; dependencyCheck=0; autowireCandidate=true; primary=false; factoryBeanName=serviceConfig; factoryMethodName=jacksonJsonProvider; initMethodName=null; destroyMethodName=(inferred); defined in class path resource [com/capgemini/poc/devonita/general/service/impl/config/ServiceConfig.class]] with [Root bean: class [null]; scope=; abstract=false; lazyInit=false; autowireMode=3; dependencyCheck=0; autowireCandidate=true; primary=false; factoryBeanName=io.oasp.module.cxf.common.impl.server.rest.CxfServerRestAutoConfiguration; factoryMethodName=jacksonJsonProvider; initMethodName=null; destroyMethodName=(inferred); defined in class path resource [io/oasp/module/cxf/common/impl/server/rest/CxfServerRestAutoConfiguration.class]]]  
[D: 2018-04-06 16:46:50,368] [P: INFO ] [C: ] [T: localhost-startStop-1] [L: org.springframework.beans.factory.support.DefaultListableBeanFactory] - [M: Overriding bean definition for bean 'restServiceExceptionFacade' with a different definition: replacing [Root bean: class [null]; scope=; abstract=false; lazyInit=false; autowireMode=3; dependencyCheck=0; autowireCandidate=true; primary=false; factoryBeanName=serviceConfig; factoryMethodName=restServiceExceptionFacade; initMethodName=null; destroyMethodName=(inferred); defined in class path resource [com/capgemini/poc/devonita/general/service/impl/config/ServiceConfig.class]] with [Root bean: class [null]; scope=; abstract=false; lazyInit=false; autowireMode=3; dependencyCheck=0; autowireCandidate=true; primary=false; factoryBeanName=io.oasp.module.cxf.common.impl.server.rest.CxfServerRestAutoConfiguration; factoryMethodName=restServiceExceptionFacade; initMethodName=null; destroyMethodName=(inferred); defined in class path resource [io/oasp/module/cxf/common/impl/server/rest/CxfServerRestAutoConfiguration.class]]]  
[D: 2018-04-06 16:46:50,743] [P: INFO ] [C: ] [T: localhost-startStop-1] [L: org.springframework.integration.config.IntegrationRegistrar] - [M: No bean named 'integrationHeaderChannelRegistry' has been explicitly defined. Therefore, a default DefaultHeaderChannelRegistry will be created.]  
[D: 2018-04-06 16:46:51,879] [P: DEBUG] [C: ] [T: localhost-startStop-1] [L: org.apache.cxf.common.logging.LogUtils] - [M: Using org.apache.cxf.common.logging.Slf4jLogger for logging.]  
[D: 2018-04-06 16:46:51,895] [P: INFO ] [C: ] [T: localhost-startStop-1] [L: org.springframework.integration.config.DefaultConfiguringBeanFactoryPostProcessor] - [M: No bean named 'errorChannel' has been explicitly defined. Therefore, a default PublishSubscribeChannel will be created.]  
[D: 2018-04-06 16:46:51,921] [P: INFO ] [C: ] [T: localhost-startStop-1] [L: org.springframework.integration.config.DefaultConfiguringBeanFactoryPostProcessor] - [M: No bean named 'taskScheduler' has been explicitly defined. Therefore, a default ThreadPoolTaskScheduler will be created.]  
[D: 2018-04-06 16:46:51,951] [P: INFO ] [C: ] [T: localhost-startStop-1] [L: org.springframework.beans.factory.annotation.AutowiredAnnotationBeanPostProcessor] - [M: JSR-330 'javax.inject.Inject' annotation found and supported for autowiring]  
[D: 2018-04-06 16:46:52,182] [P: WARN ] [C: ] [T: localhost-startStop-1] [L: org.springframework.boot.context.embedded.AnnotationConfigEmbeddedWebApplicationContext] - [M: Exception encountered during context initialization - cancelling refresh attempt: org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'org.springframework.ws.config.annotation.DelegatingWsConfiguration': Unsatisfied dependency expressed through method 'setConfigurers' parameter 0; nested exception is org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'serviceConfig': Unsatisfied dependency expressed through field 'objectMapperFactory'; nested exception is org.springframework.beans.factory.NoSuchBeanDefinitionException: No qualifying bean of type 'io.oasp.module.rest.service.impl.json.ObjectMapperFactory' available: expected at least 1 bean which qualifies as autowire candidate. Dependency annotations: {@javax.inject.Inject()}]  
[D: 2018-04-06 16:46:52,190] [P: ERROR] [C: ] [T: localhost-startStop-1] [L: org.springframework.beans.factory.support.DefaultListableBeanFactory] - [M: Destroy method on bean with name 'org.springframework.boot.context.properties.ConfigurationPropertiesBindingPostProcessor' threw an exception]  
java.lang.IllegalStateException: ApplicationEventMulticaster not initialized - call 'refresh' before multicasting events via the context: org.springframework.boot.context.embedded.AnnotationConfigEmbeddedWebApplicationContext@460a8c06: startup date [Fri Apr 06 16:46:47 CEST 2018]; root of context hierarchy
         at org.springframework.context.support.AbstractApplicationContext.getApplicationEventMulticaster(AbstractApplicationContext.java:404)
         at org.springframework.context.support.ApplicationListenerDetector.postProcessBeforeDestruction(ApplicationListenerDetector.java:97)
         at org.springframework.beans.factory.support.DisposableBeanAdapter.destroy(DisposableBeanAdapter.java:253)
         at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.destroyBean(DefaultSingletonBeanRegistry.java:578)
         at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.destroySingleton(DefaultSingletonBeanRegistry.java:554)
         at org.springframework.beans.factory.support.DefaultListableBeanFactory.destroySingleton(DefaultListableBeanFactory.java:961)
         at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.destroySingletons(DefaultSingletonBeanRegistry.java:523)
         at org.springframework.beans.factory.support.DefaultListableBeanFactory.destroySingletons(DefaultListableBeanFactory.java:968)
         at org.springframework.context.support.AbstractApplicationContext.destroyBeans(AbstractApplicationContext.java:1033)
         at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:555)
         at org.springframework.boot.context.embedded.EmbeddedWebApplicationContext.refresh(EmbeddedWebApplicationContext.java:122)
         at org.springframework.boot.SpringApplication.refresh(SpringApplication.java:737)
         at org.springframework.boot.SpringApplication.refreshContext(SpringApplication.java:370)
         at org.springframework.boot.SpringApplication.run(SpringApplication.java:314)
         at org.springframework.boot.web.support.SpringBootServletInitializer.run(SpringBootServletInitializer.java:151)
         at org.springframework.boot.web.support.SpringBootServletInitializer.createRootApplicationContext(SpringBootServletInitializer.java:131)
         at org.springframework.boot.web.support.SpringBootServletInitializer.onStartup(SpringBootServletInitializer.java:86)
         at org.springframework.web.SpringServletContainerInitializer.onStartup(SpringServletContainerInitializer.java:169)
         at org.apache.catalina.core.StandardContext.startInternal(StandardContext.java:5244)
         at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:147)
         at org.apache.catalina.core.ContainerBase$StartChild.call(ContainerBase.java:1408)
         at org.apache.catalina.core.ContainerBase$StartChild.call(ContainerBase.java:1398)
         at java.util.concurrent.FutureTask.run(FutureTask.java:266)
         at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
         at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
         at java.lang.Thread.run(Thread.java:748)
[D: 2018-04-06 16:46:52,192] [P: ERROR] [C: ] [T: localhost-startStop-1] [L: org.springframework.beans.factory.support.DefaultListableBeanFactory] - [M: Destroy method on bean with name 'org.springframework.boot.autoconfigure.internalCachingMetadataReaderFactory' threw an exception]  
java.lang.IllegalStateException: ApplicationEventMulticaster not initialized - call 'refresh' before multicasting events via the context: org.springframework.boot.context.embedded.AnnotationConfigEmbeddedWebApplicationContext@460a8c06: startup date [Fri Apr 06 16:46:47 CEST 2018]; root of context hierarchy
         at org.springframework.context.support.AbstractApplicationContext.getApplicationEventMulticaster(AbstractApplicationContext.java:404)
         at org.springframework.context.support.ApplicationListenerDetector.postProcessBeforeDestruction(ApplicationListenerDetector.java:97)
         at org.springframework.beans.factory.support.DisposableBeanAdapter.destroy(DisposableBeanAdapter.java:253)
         at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.destroyBean(DefaultSingletonBeanRegistry.java:578)
         at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.destroySingleton(DefaultSingletonBeanRegistry.java:554)
         at org.springframework.beans.factory.support.DefaultListableBeanFactory.destroySingleton(DefaultListableBeanFactory.java:961)
         at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.destroySingletons(DefaultSingletonBeanRegistry.java:523)
         at org.springframework.beans.factory.support.DefaultListableBeanFactory.destroySingletons(DefaultListableBeanFactory.java:968)
         at org.springframework.context.support.AbstractApplicationContext.destroyBeans(AbstractApplicationContext.java:1033)
         at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:555)
         at org.springframework.boot.context.embedded.EmbeddedWebApplicationContext.refresh(EmbeddedWebApplicationContext.java:122)
         at org.springframework.boot.SpringApplication.refresh(SpringApplication.java:737)
         at org.springframework.boot.SpringApplication.refreshContext(SpringApplication.java:370)
         at org.springframework.boot.SpringApplication.run(SpringApplication.java:314)
         at org.springframework.boot.web.support.SpringBootServletInitializer.run(SpringBootServletInitializer.java:151)
         at org.springframework.boot.web.support.SpringBootServletInitializer.createRootApplicationContext(SpringBootServletInitializer.java:131)
         at org.springframework.boot.web.support.SpringBootServletInitializer.onStartup(SpringBootServletInitializer.java:86)
         at org.springframework.web.SpringServletContainerInitializer.onStartup(SpringServletContainerInitializer.java:169)
         at org.apache.catalina.core.StandardContext.startInternal(StandardContext.java:5244)
         at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:147)
         at org.apache.catalina.core.ContainerBase$StartChild.call(ContainerBase.java:1408)
         at org.apache.catalina.core.ContainerBase$StartChild.call(ContainerBase.java:1398)
         at java.util.concurrent.FutureTask.run(FutureTask.java:266)
         at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
         at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
         at java.lang.Thread.run(Thread.java:748)
[D: 2018-04-06 16:46:52,696] [P: ERROR] [C: ] [T: localhost-startStop-1] [L: org.springframework.boot.diagnostics.LoggingFailureAnalysisReporter] - [M: 

***************************
APPLICATION FAILED TO START
***************************

Description:

Field objectMapperFactory in com.capgemini.poc.devonita.general.service.impl.config.ServiceConfig required a bean of type 'io.oasp.module.rest.service.impl.json.ObjectMapperFactory' that could not be found.


Action:

Consider defining a bean of type 'io.oasp.module.rest.service.impl.json.ObjectMapperFactory' in your configuration.
]  
[D: 2018-04-06 16:46:52,699] [P: ERROR] [C: ] [T: Catalina-startStop-1] [L: org.apache.catalina.core.ContainerBase] - [M: A child container failed during start]  
java.util.concurrent.ExecutionException: org.apache.catalina.LifecycleException: Failed to start component [StandardEngine[Catalina].StandardHost[localhost].StandardContext[/devonita-server]]
         at java.util.concurrent.FutureTask.report(FutureTask.java:122)
         at java.util.concurrent.FutureTask.get(FutureTask.java:192)
         at org.apache.catalina.core.ContainerBase.startInternal(ContainerBase.java:916)
         at org.apache.catalina.core.StandardHost.startInternal(StandardHost.java:871)
         at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:147)
         at org.apache.catalina.core.ContainerBase$StartChild.call(ContainerBase.java:1408)
         at org.apache.catalina.core.ContainerBase$StartChild.call(ContainerBase.java:1398)
         at java.util.concurrent.FutureTask.run(FutureTask.java:266)
         at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
         at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
         at java.lang.Thread.run(Thread.java:748)
Caused by: org.apache.catalina.LifecycleException: Failed to start component [StandardEngine[Catalina].StandardHost[localhost].StandardContext[/devonita-server]]
         at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:153)
         ... 6 common frames omitted
```

It seems that the problem is about the generating of SOAP  or SOAP and REST.

--- a/cobigen-templates/templates-oasp4j/crud_java_server_app/templates/java/${variables.rootPackage}/${variables.component}/logic/api/to/${variables.entityName}Eto.java.ftl
+++ b/cobigen-templates/templates-oasp4j/crud_java_server_app/templates/java/${variables.rootPackage}/${variables.component}/logic/api/to/${variables.entityName}Eto.java.ftl
@@ -10,7 +10,7 @@ import java.util.Set;
 /**
  * Entity transport object of ${variables.entityName}
  */
-public class ${variables.entityName}Eto extends <#if pojo.extendedType.canonicalName=="java.lang.Object" || pojo.extendedType.package!=pojo.package>AbstractEto<#else>${pojo.extendedType.name?replace("Entity","Eto")}</#if> implements ${variables.entityName} {
+public class ${variables.entityName}Eto extends <#if pojo.extendedType.canonicalName=="java.lang.Object" || pojo.extendedType.package!=pojo.package || pojo.extendedType.name == "ApplicationPersistenceEntity">AbstractEto<#else>${pojo.extendedType.name?replace("Entity","Eto")}</#if> implements ${variables.entityName} {
 
 	private static final long serialVersionUID = 1L;
 

--- a/cobigen-templates/templates-oasp4j/crud_java_server_app/templates/java/${variables.rootPackage}/general/service/impl/config/ServiceConfig.java.ftl
+++ b/cobigen-templates/templates-oasp4j/crud_java_server_app/templates/java/${variables.rootPackage}/general/service/impl/config/ServiceConfig.java.ftl
@@ -28,7 +28,7 @@ import com.fasterxml.jackson.jaxrs.json.JacksonJsonProvider;
 import ${variables.rootPackage}.${variables.component}.service.impl.soap.${variables.component?cap_first}SoapServiceImpl;
 
 import io.oasp.module.rest.service.impl.RestServiceExceptionFacade;
-import io.oasp.module.rest.service.impl.json.ObjectMapperFactory;
+import io.oasp.module.json.common.base.ObjectMapperFactory;
 
 /**
  * {@link Configuration} for (REST or SOAP) services using CXF.

